src/objectExplorer/objectExplorerService.ts

```diff
@@ -293,6 +293,10 @@ export class ObjectExplorerService {
 
     async getChildren(element?: TreeNodeInfo): Promise<vscode.TreeItem[]> {
         if (element) {
+            // set current node for very first expansion of disconnected node
```

Comment:
So the comment isn't really accurate since it'll do this for every single node that children are requested of. So would be preferable to update that.

But in general - how is this fixing the issue? You set the currentNode whenever a new item is selected in the tree - is that not happening when expanding a node? I would expect it would since it does actually select the node it expands.

Reply:
@Charles-Gagnon The issue is that it does set the current node, but that's after the `getChildren` call, which needs its value.

Reply:
Where is it using the currentNode that's failing? You have the "current" node already (element param) so why wouldn't you just be using that like the rest of the getChildren method is doing?

Reply:
@Charles-Gagnon That's exactly the issue. When a node with a command is clicked or right clicked for context menu, we get the node parameter. This is the case where there's a collapsed disconnected server, which when clicked on goes to the `getChildren` call from the provider. 

Now, when there's a failure (firewall rule in this case), it goes here - https://github.com/microsoft/vscode-mssql/pull/1614/files#diff-9401975d07f4df26ce1df2d776b8c746R105

Since object explorer gives notifications about sessions instead of callbacks. Now, at this point, the current node isn't set, and hence this fails. The fix addresses this issue.